ksql-engine/src/test/java/io/confluent/ksql/statement/PropertyOverrideInjectorTest.java

```diff
@@ -13,62 +13,68 @@
  * specific language governing permissions and limitations under the License.
  */
 
-package io.confluent.ksql.rest.server.validation;
+package io.confluent.ksql.statement;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.not;
 
 import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
 import io.confluent.ksql.parser.tree.SetProperty;
 import io.confluent.ksql.parser.tree.UnsetProperty;
-import io.confluent.ksql.rest.server.TemporaryEngine;
-import io.confluent.ksql.statement.ConfiguredStatement;
 import io.confluent.ksql.util.KsqlConfig;
 import io.confluent.ksql.util.KsqlStatementException;
 import java.util.HashMap;
+import java.util.Map;
 import java.util.Optional;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
+import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
-@RunWith(MockitoJUnitRunner.class)
-public class PropertyValidatorTest {
+public class PropertyOverrideInjectorTest {
 
-  @Rule public final TemporaryEngine engine = new TemporaryEngine();
   @Rule public final ExpectedException expectedException = ExpectedException.none();
 
+  private final KsqlConfig ksqlConfig = new KsqlConfig(new HashMap<>());
+
```

Comment:
we should add a test case that exercises the validation done by `cloneWithPropertyOverrides`

Reply:
It does 🎉  `shouldFailOnInvalidSetPropertyValue` and `shouldFailOnUnknownUnsetProperty`